packages/postgres-database/src/simcore_postgres_database/utils_folders_v2.py

```diff
@@ -0,0 +1,916 @@
+    )
+    MOVE_FOLDER: ClassVar[_FolderPermissions] = _or_dicts_list(
+        [_MOVE_FOLDER_SOURCE, _MOVE_FOLDER_DESTINATION]
+    )
+
+
+NO_ACCESS_PERMISSIONS: _FolderPermissions = _make_permissions()
+
+VIEWER_PERMISSIONS: _FolderPermissions = _or_dicts_list(
+    [
+        _BasePermissions.LIST_FOLDERS,
+    ]
+)
+EDITOR_PERMISSIONS: _FolderPermissions = _or_dicts_list(
+    [
+        VIEWER_PERMISSIONS,
+        _BasePermissions.CREATE_FOLDER,
+        _BasePermissions.ADD_PROJECT_TO_FOLDER,
+    ]
+)
+OWNER_PERMISSIONS: _FolderPermissions = _or_dicts_list(
+    [
+        EDITOR_PERMISSIONS,
+        _BasePermissions.SHARE_FOLDER,
+        _BasePermissions.UPDATE_FODLER,
+        _BasePermissions.DELETE_FOLDER,
+        _BasePermissions.REMOVE_PROJECT_FROM_FOLDER,
+        _BasePermissions.MOVE_FOLDER,
+    ]
+)
+
+_ROLE_TO_PERMISSIONS: dict[FolderAccessRole, _FolderPermissions] = {
+    FolderAccessRole.NO_ACCESS: NO_ACCESS_PERMISSIONS,
+    FolderAccessRole.VIEWER: VIEWER_PERMISSIONS,
+    FolderAccessRole.EDITOR: EDITOR_PERMISSIONS,
+    FolderAccessRole.OWNER: OWNER_PERMISSIONS,
+}
+
+
+def _hash_permissions(permissions: _FolderPermissions) -> tuple:
+    return tuple(permissions.items())
+
+
+_PERMISSIONS_TO_ROLE: dict[tuple, FolderAccessRole] = {
+    _hash_permissions(NO_ACCESS_PERMISSIONS): FolderAccessRole.NO_ACCESS,
+    _hash_permissions(VIEWER_PERMISSIONS): FolderAccessRole.VIEWER,
+    _hash_permissions(EDITOR_PERMISSIONS): FolderAccessRole.EDITOR,
+    _hash_permissions(OWNER_PERMISSIONS): FolderAccessRole.OWNER,
+}
+
+
+def _get_permissions_from_role(role: FolderAccessRole) -> _FolderPermissions:
+    return _ROLE_TO_PERMISSIONS[role]
+
+
+def _requires(*permissions: _FolderPermissions):
+    if len(permissions) == 0:
+        return _make_permissions()
+    return _or_dicts_list(permissions)
+
+
+def _get_true_permissions(
+    permissions: _FolderPermissions, table
+) -> ColumnElement | bool:
+    """compose SQL where clause where only for the entries that are True"""
+    clauses: list[ColumnElement] = []
+
+    if permissions["read"]:
+        clauses.append(table.c.read.is_(True))
+    if permissions["write"]:
+        clauses.append(table.c.write.is_(True))
+    if permissions["delete"]:
+        clauses.append(table.c.delete.is_(True))
+
+    return sa.and_(*clauses) if clauses else True
+
+
+def _get_all_permissions(permissions: _FolderPermissions, table) -> ColumnElement:
+    return sa.and_(
+        table.c.read.is_(permissions["read"]),
+        table.c.write.is_(permissions["write"]),
+        table.c.delete.is_(permissions["delete"]),
+    )
+
+
+###
+### UTILS NAMING
+###
+
+_RE_FOLDER_NAME_INVALID_CHARS: Final[str] = r'[<>:"/\\|?*]'
+_FOLDER_NAMES_RESERVED_WINDOWS: Final[set[str]] = {
+    "CON",
+    "PRN",
+    "AUX",
+    "NUL",
+    *[f"COM{i}" for i in range(1, 10)],
+    *[f"LPT{i}" for i in range(1, 10)],
+}
+_FOLDER_NAME_MAX_LENGTH: Final[NonNegativeInt] = 255
+
```

Comment:
at this point, why you do not create pydantic models instead of re-inventing the whole validation?

Reply:
Spent 30 minutes trying to make this easy one work. Had no success at all at making a validator that works.

Reply:
I'm unsure what I'm doing wrong.

Reply:
done, managed to replace it.